### ISSUE TITLE
Probe table header scope issue

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#8152]
# Story

As a Digital Accessibility Coordinator, I would like to see SiteNow tables include the `scope` attribute on table headers without content editors having to manually apply the html.

# Additional Info

This may have been a regression introduced when upgrading to ckeditor 5.

A couple upstream issues:
https://www.drupal.org/project/drupal/issues/3427076
https://github.com/ckeditor/ckeditor5/issues/3175

A possible related issue in our backlog:
https://github.com/uiowa/uiowa/issues/7440

Should we probe this to see if there is anything we can do to help the upstream issues or to resolve this for SiteNow while we wait for solutions upstream?

**Comment by briand44** [comment]
Grooming conversation... We should probe to see if there is a fairly straight forward way of outputting the correct `scope` attribute for table headers for SiteNow sites. If there isn't a solution for SiteNow sites and we need to wait for upstream issues we should create a known issue on the SiteNow site and comment on upstream issues.

**Comment by pyrello** [comment]
I took a look at the current state of play upstream of us. 

First, I read through the upstream Drupal issues I could find:
- https://www.drupal.org/project/drupal/issues/3427076
- https://www.drupal.org/project/drupal/issues/3384400
- https://www.drupal.org/project/drupal/issues/3340578#comment-15313765

From these, it appears that there is no work going on in Drupal core currently to fix these issues. Rather, they have been marked as `[Upstream]` and are waiting for the CKEditor team to fix them. The last issue linked above is a list of CKEditor5 issues that have been prioritized from Drupal core's standpoint as most important. That includes an issue that has been identified as possibly overlapping with the scope issue, but it doesn't seem to have ever been definitively linked as the solution. 

A comment in the first issue thread contains a somewhat vague description of a workaround put in place for a project to get around this issue. While I'm not sure there is enough detail in the description to fully recreate it, it is possible we could reach out to the commenter and see if they could provide a code example.

Next, I took a look at CKE5 issues and looked for any evidence of a code solution in progress:
- https://github.com/ckeditor/ckeditor5/issues/3175
- https://github.com/ckeditor/ckeditor5/issues/17185 (Marked a duplicate of the previous one)
- https://github.com/ckeditor/ckeditor5/issues/14911

The last issue is the issue identified as a priority for Drupal core and possibly the one that will fix the issue with scope attributes not being generated, although that is not totally clear yet.

I could only find the following PR containing possible code changes as evidence of a solution being worked on:
- https://github.com/ckeditor/ckeditor5/pull/15366

The last activity on that PR is from 11/2023, so it seems like there is no active work going on currently to fix this issue.

Based on the above, next steps would be:
- Try to get a code example from the comment in the Drupal issue above and see if that example can be adapted to our purposes.
- Attempt to roll our own solution fixing this as a CKEditor plugin.

**Comment by pyrello** [comment]
> Try to get a code example from the comment in the Drupal issue above and see if that example can be adapted to our purposes.

I went ahead and sent a message to the user to see if they will provide a code example or provide more details of their solution.

**Comment by pyrello** [comment]
Going to try this out today: https://www.drupal.org/project/table_header_scope_attribute. This was sent to me by the person I emailed on Friday.

**Comment by pyrello** [comment]
The table_header_scope_attribute module does appear to work to fix the issue of the scopes not appearing for header rows and columns. However, as an input filter, it runs before the HTML is rendered rather than acting on the HTML when it is generated in the WYSIWYG text area. That means that it is not visible in source. If someone enters something in source it appears to get overridden by the input filter. Not a perfect solution, but it probably works well enough to get this off our plates until a better solution comes along.

There are going to be some issues with getting the module installed locally. I tested it using simplytest.me because I could not easily get it set up in a local SiteNow environment. We will need to update our filter config and we will potentially need to update `docroot/modules/custom/uiowa_core/src/Plugin/Filter/FilterIframe.php`, which causes an error when trying to save a filter format for which it is enabled.

My recommendation is that we move ahead with this module as our solution, create a known issue for the fact that source code can be overridden by the filter plugin, and deal with the platform updates we'll need to make to get the module installed.
